homeassistant/components/zwave_js/diagnostics.py

```diff
@@ -13,19 +16,65 @@
 from homeassistant.helpers.aiohttp_client import async_get_clientsession
 
 from .const import DATA_CLIENT, DOMAIN
-from .helpers import get_home_and_node_id_from_device_entry
+from .helpers import ZwaveValueID, get_home_and_node_id_from_device_entry
 
-TO_REDACT = {"homeId", "location"}
+KEYS_TO_REDACT = {"homeId", "location"}
+
+VALUES_TO_REDACT = (
+    ZwaveValueID(property_="userCode", command_class=CommandClass.USER_CODE),
+)
+
+
+def redact_value_of_zwave_value(zwave_value: ValueDataType) -> ValueDataType:
+    """Redact value of a Z-Wave value."""
+    for value_to_redact in VALUES_TO_REDACT:
+        if (
+            (
+                value_to_redact.command_class is None
```

Comment:
I don't think we should match if the matcher attribute is None. Then we'll match if all matcher attributes are None too. That's weird.

I suggest we only match with the attributes that are not None in the matcher.

Reply:
good call

Reply:
OK I handled this in the `ZwaveValueID` class because the scenario you mentioned should not be valid